homeassistant/components/zwave_js/api.py

```diff
@@ -318,14 +318,43 @@ async def websocket_refresh_node_info(
     entry_id = msg[ENTRY_ID]
     node_id = msg[NODE_ID]
     client = hass.data[DOMAIN][entry_id][DATA_CLIENT]
-    node = client.driver.controller.nodes.get(node_id)
+    controller = client.driver.controller
+    node = controller.nodes.get(node_id)
+
+    @callback
+    def async_cleanup() -> None:
+        """Remove signal listeners."""
+        for unsub in unsubs:
+            unsub()
+
+    @callback
+    def forward_event(event: dict) -> None:
+        connection.send_message(
+            websocket_api.event_message(msg[ID], {"event": event["event"]})
+        )
+
+    @callback
+    def forward_stage(event: dict) -> None:
+        connection.send_message(
+            websocket_api.event_message(
+                msg[ID], {"event": event["event"], "stage": event["stageName"]}
+            )
+        )
 
     if node is None:
         connection.send_error(msg[ID], ERR_NOT_FOUND, f"Node {node_id} not found")
         return
 
-    await node.async_refresh_info()
-    connection.send_result(msg[ID])
+    connection.subscriptions[msg["id"]] = async_cleanup
+    unsubs = [
+        node.on("interview started", forward_event),
+        node.on("interview completed", forward_event),
```

Comment:
Wouldn't you also want to call `async_cleanup` here and for `interview failed` since there will be no more messages for the original command to refresh?

Reply:
Once the frontend receives those events it unsubscribes which should trigger the cleanup